### PR TITLE
Fabtest: test_rma_pingpong bugfix

### PR DESCRIPTION
**Pull request by jiaxiyan**
1. The fixture memory_type was missing in test_rma_pingpong_range_no_inject
2. The RMA pingpong test should not run with 0-byte messages as zero-sized transfers are not supported. We should use different message sizes for rma pingpong, starting from non-zero values.

### CODE REVIEW

File: fabtests/pytest/efa/test_rma_pingpong.py [hunk 1]
```diff
@@ -31,7 +36,7 @@ def test_rma_pingpong_range(cmdline_args, operation_type, completion_semantic, m
 
 @pytest.mark.functional
 @pytest.mark.parametrize("operation_type", ["writedata", "write"])
-def test_rma_pingpong_range_no_inject(cmdline_args, operation_type, completion_semantic, inject_message_size):
+def test_rma_pingpong_range_no_inject(cmdline_args, operation_type, completion_semantic, inject_message_size, memory_type):
```
Comment (wenduwan) [inline fabtests/pytest/efa/test_rma_pingpong.py:39]: Does this test work for 0 B message?

File: fabtests/pytest/efa/test_rma_pingpong.py [hunk 2]
```diff
@@ -19,6 +19,11 @@ def test_rma_pingpong(cmdline_args, iteration_type, operation_type, completion_s
 
 @pytest.mark.functional
 @pytest.mark.parametrize("operation_type", ["writedata", "write"])
+@pytest.mark.parametrize("message_size", ["r:1,4,65",
+                                          "r:4048,4,4148",
+                                          "r:8000,4,9000",
+                                          "r:17000,4,18000",
+                                          "r:1,1024,1048577"])
```
Comment (wenduwan) [inline fabtests/pytest/efa/test_rma_pingpong.py:26]: I'm not sure if we want `r:1,4,65` - it expands to `1,5,9,13,...` which does not look right.

Overall I think it's probably easier to skip the test if message size starts from 0.

Comment (jiaxiyan) [reply fabtests/pytest/efa/test_rma_pingpong.py:26]: so we just test "r:4048,4,4148",
                          "r:8000,4,9000",
                          "r:17000,4,18000"?

**Review by wenduwan — APPROVED** [review]

**Comment by shijin-aws** [comment]
Can you add this test to AWS CI so we can test the change? @a-szegel tried to add it earlier but hit similar issue that this PR is going to address https://github.com/ofiwg/libfabric/pull/9847

**Comment by shijin-aws** [comment]
Please check AWS CI failure

**Comment by jiaxiyan** [comment]
test_rma_pingpong_range  error message is 

```
 Client timed out
 libfabric:172391:1717730028::efa:ep_ctrl:efa_rdm_ep_destroy_buffer_pools():713<warn> Closing ep with unreleased rxe
 Data check error (D !=) at byte 17771 for buffer size 17775
```

Not sure how this change affects test_fabtests_functional[non-dmabuf]

**Comment by wenduwan** [comment]
@jiaxiyan Were you able to find the root cause of the timeout?

**Comment by wenduwan** [comment]
To run the test we also need to add the pytest file to https://github.com/ofiwg/libfabric/blob/main/fabtests/Makefile.am

File: fabtests/pytest/efa/test_rma_pingpong.py [hunk 3]
```diff
@@ -3,39 +3,40 @@
...
                          [pytest.param("short", marks=pytest.mark.short),
                           pytest.param("standard", marks=pytest.mark.standard)])
 def test_rma_pingpong(cmdline_args, iteration_type, operation_type, completion_semantic, memory_type):
-    if memory_type != "host_to_host" and operation_type == "write":
-        pytest.skip("no hmem memory support for pingpong_rma write test")
     command = "fi_rma_pingpong -e rdm"
     command = command + " -o " + operation_type + " " + perf_progress_model_cli
     # rma_pingpong test with data verification takes longer to finish
-    timeout = max(540, cmdline_args.timeout)
+    timeout = max(1020, cmdline_args.timeout)
```
Comment (wenduwan) [inline fabtests/pytest/efa/test_rma_pingpong.py:21]: Based on my test this timeout is too long. Suggest removing the max.

File: fabtests/benchmarks/benchmark_shared.c [hunk 4]
```diff
@@ -158,19 +158,21 @@ int pingpong_rma(enum ft_rma_opcodes rma_op, struct fi_rma_iov *remote)
 	if (ft_check_opts(FT_OPT_VERIFY_DATA))
 		inject_size = 0;
 
-	ret = ft_sync();
-	if (ret)
-		return ret;
-
```
Comment (shijin-aws) [inline fabtests/benchmarks/benchmark_shared.c:164]: I missed the context of this change. Why do we need such commit? @wenduwan

Comment (shijin-aws) [reply fabtests/benchmarks/benchmark_shared.c:164]: Are you saying that fi_write can happen before that overwrite is done on receiver side, if we don't make the overwrite before sync?

Comment (wenduwan) [reply fabtests/benchmarks/benchmark_shared.c:164]: Yes. For example, consider client(sender) and server(receiver):
- Receiver posts receive(RX1)
- Sender sends msg with delivery complete(TX1)
- Receiver gets receive completion(RX1)
- Receiver sends reply with delivery complete(TX2)
- Sender gets send completion(TX1)
- Sender posts receive(RX2)
- Sender gets receive completion(RX2)
- Sender starts fi_write
- Receive RX buffer modified by fi_write
- Receiver gets send completion(TX2)
- Receiver modifies RX buffer

**Comment by shijin-aws** [comment]
Intel CI doesn't run fabtests'pytest. Ignoring.

**Pull request merged**